src/openfermion/utils/_special_operators.py

```diff
@@ -309,3 +309,23 @@ def number_operator(n_orbitals, orbital=None, coefficient=1.):
     else:
         operator = FermionOperator(((orbital, 1), (orbital, 0)), coefficient)
     return operator
+
+
+def boson_number_operator(n_modes, mode=None, coefficient=1.):
```

Comment:
I am thinking perhaps we should have one function with a switch between bosons and fermions. Just a thought.

Reply:
I agree - should I merge them into a single number_operator?

Reply:
sure, thanks!